ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp

```diff
@@ -26,14 +26,27 @@ AllGather create_all_gather_struct(
     uint32_t num_devices = devices.size();
 
     uint32_t device_index = 0; // Initialize device index
-    uint32_t receiver_device_id = 0; // Initialize receiver device ID
-    uint32_t sender_device_id = 0; // Initialize sender device ID
+    std::optional<uint32_t> receiver_device_id = std::nullopt; // Initialize receiver device ID
+    std::optional<uint32_t> sender_device_id = std::nullopt; // Initialize sender device ID
 
     for (uint32_t i = 0; i < num_devices; ++i) {
         if (devices[i] == input_tensor.device()) {
             device_index = i;
-            receiver_device_id = devices[(i + 1) % num_devices]->id(); // Next device in the ring
-            sender_device_id = devices[(i + num_devices - 1) % num_devices]->id(); // Previous device in the ring
+            if (topology == ttnn::ccl::Topology::Ring) {
```

Comment:
Just a thought. Change is not required, but I recommend to consider. 

Using switch instead of ifs here would be best. When a new enum value added to the enum type, this code will stop compiling till the new value is handled, making it less error prone.